gcpdiag/lint/gke/bp_2021_002_groups_enabled.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Google Groups for RBAC enabled.
```

Comment:
I feel like this rule might be triggered too frequently and be possibly annoying for gcpdiag users (I know for instance that I personally rarely enable this). Is it really the case that we always recommend to use Google Groups for RBAC? I am not sure, especially also considering that even Autopilot doesn't have it enabled. I'll ask the opinion of some more folks on this.

Reply:
Adding in a temporary condition that this check won't apply to Autopilot clusters makes sense and can be added.

I don't think we should be too concerned about the rule being triggered too frequently because users can skip categories of lint rules as needed. 

I think it's worth adding the check for two reasons:
1. If we use gke/BP/2021_001 as an example, I know of scenarios why that wouldn't be enabled on GKE clusters just like Groups for RBAC. However, I would recommend both for production clusters far more often than not for reasons outlined in the documentation.
2. Groups for RBAC on GKE requires extra setup that isn't always required on other clouds, to enable group permission management specifically, so the check can be a good learning opportunity for new administrators coming into GCP.

Reply:
gke/BP/2021_001 is not comparable in my opinion, because we definitely always recommend to enable Cloud Ops monitoring+logging, at least for system workloads. One guiding principle is that we are going to recommend everything that improves supportability of a product. Another example for this is recommending to enable serial output logging for GCE instances. There should be rarely a reason not to fulfill this best-practice.

Groups for RBAC is a complex feature that requires access at the organizational level, which I think most GKE administrators don't have (I am actually struggling right now to create a cluster where you could test this feature). I am still gathering feedback on it, but in my opinion this is going to be generating too many positives. In the end, best-practice rules are going to be opinionated by nature.

It is true that it is possible to exclude a class of rules or this one specifically, but every time that we implement a new rule it is always a tradeoff between the potential of raising too many issues that the user doesn't care about, and raising something that they might be interested in.

I wonder whether we should have a new class of rules that are disabled by default, like "HINT" or something like that. Rules that raise things that might be interesting for you, especially to learn about things. What do you think?

Reply:
We do have that now: BP_EXT is for best-practice rules that are disabled by default.

Reply:
Updated to BP_EXT

Reply:
Thanks Taylor. There is a code conflict that remains, but otherwise I think this is ready to be merged.

Reply:
Merge conflict should be resolved now.